### ISSUE TITLE
New issue found : Youtube player keep loading only after dragged the seek bar , its unable to play video.

### ISSUE THREAD

**Issue by amitimg29** [PierfrancescoSoffritti/android-youtube-player#734]

**Comment by videopromoter** [comment]
Facing similar issues, when using seekTo method, the video gets stuck in Buffering state.

**Comment by amitimg29** [comment]
is there any solution available?

**Comment by singhaayush** [comment]
I have observed that if we call loadVideo(videoid,seekedPosition) for the first time when user seeks player it fixes the issue of infinite buffering player state  couldn't find the actual cause .

**Comment by sonusourav** [comment]
I don't think the issue is from the youtube api's because in mine case, when I am using custom controller then only I am facing this and when using the sample app provided in this repo, then I am not facing.
My custom controller is almost same as the default one in this repo.

**Comment by sonusourav** [comment]
> 
> 
> I have observed that if we call loadVideo(videoid,seekedPosition) for the first time when user seeks player it fixes the issue of infinite buffering player state couldn't find the actual cause .

But isn't it a hack and not a fix. Also, since it loads the video again, it does not give a smooth experience

**Comment by singhaayush** [comment]
> > I have observed that if we call loadVideo(videoid,seekedPosition) for the first time when user seeks player it fixes the issue of infinite buffering player state couldn't find the actual cause .
> 
> But isn't it a hack and not a fix. Also, since it loads the video again, it does not give a smooth experience

Yes it is not right approach . I m facing this issue with certain set of videos even in the sample app of library itself .

**Comment by sonusourav** [comment]
Hey @PierfrancescoSoffritti Could you please look into this once?

**Comment by PierfrancescoSoffritti** [comment]
Hey, the code is open source, feel free to take a look. This issue never happened to me, which makes me think that could be an error on your side, or YouTube behaving differently in your region.

**Comment by SlayerMMD** [comment]
Hey all,
Just wanted to be sure. Those who all are facing issues are using the native player or the webui based one?
This will help us track down where the actual issue might be

**Comment by singhaayush** [comment]
> Hey all,
> Just wanted to be sure. Those who all are facing issues are using the native player or the webui based one?
> This will help us track down where the actual issue might be

I  am using custom uicontroller (native ui) but the issue can be easly reproduced on both i checked in sample app with this video 
videoId 6Ogi2gn92Lk .   I have also tried to run the exact iframe html on pc web browser the player is working fine there . But same html is causing issues when played inside android webview .

**Comment by sonusourav** [comment]
Yeah, for me the issue is occurring in both native as well as webui.
Other points that I noted are following:
- I have tried using the library in a separate new project and it works completely fine. Both the default and the custom version are working fine.
- The weird thing is I copy pasted exactly code from my current project to the new project and in the new project it is working fine while in my current project, its not.
- Initially I thought, the problem is with the custom controller that I am using in my current project, but without using it also I am facing the same.

**Comment by sonusourav** [comment]
The way I am implementing it is, I am showing the youtube player in a pop up window

### MainActivity.kt
```
       
        val ytLayout = View.inflate(activity, R.layout.dialog_youtube_player, null)     // The UI that will be on the pop up screen
        player = ytLayout.findViewById(R.id.youtubePlayer)                                    // The youtubePlayerView
        activity.lifecycle.addObserver(player)

        val width = getScreenWidth()
        val window = PopupWindow(
                ytLayout,
                width - 50,
                ViewGroup.LayoutParams.WRAP_CONTENT,
                true
        )
        window.setBackgroundDrawable(ColorDrawable(Color.WHITE))
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.elevation = 20f
        }

     //Using delay because activity ui might not be ready
        Handler(Looper.getMainLooper()).postDelayed({
            window.showAtLocation(ytLayout, Gravity.TOP, mPosX, mPosY)
        }, 100)


        val customPlayerUi: View = player.inflateCustomPlayerUi(R.layout.custom_youtube_player)
        player.addYouTubePlayerListener(object : AbstractYouTubePlayerListener() {
            override fun onReady(youTubePlayer: YouTubePlayer) {
                customYoutubeUiController = CustomYoutubeUiController(customPlayerUi, youTubePlayer)
                youTubePlayer.addListener(customYoutubeUiController!!)
                youTubePlayer.loadOrCueVideo(activity.lifecycle, videoId, 0f)
            }
        })
```


### CustomYoutubeUiController

```
class CustomYoutubeUiController(controlsView: View, private val youTubePlayer: YouTubePlayer)
    :  AbstractYouTubePlayerListener(), YouTubePlayerSeekBarListener {

  
    private val panel: View = controlsView.findViewById(R.id.panel)
    private val controlsContainer: View = controlsView.findViewById(R.id.controls_container)
    private val progressBar: ProgressBar = controlsView.findViewById(R.id.progress)
    private val playPauseButton: ImageView = controlsView.findViewById(R.id.play_pause_button)
    private val youtubePlayerSeekBar: YouTubePlayerSeekBar = controlsView.findViewById(R.id.youtube_player_seekbar)
    private val fadeControlsContainer: FadeViewHelper = FadeViewHelper(controlsContainer)

    private var isPlaying = false
    private var isPlayPauseButtonEnabled = true

    init {
        initClickListeners()
    }

    private fun initClickListeners() {
        youTubePlayer.addListener(youtubePlayerSeekBar)
        youTubePlayer.addListener(fadeControlsContainer)

        youtubePlayerSeekBar.youtubePlayerSeekBarListener = this
        panel.setOnClickListener { fadeControlsContainer.toggleVisibility() }
        playPauseButton.setOnClickListener { onPlayButtonPressed() }
    }

    override fun seekTo(time: Float) {
        youTubePlayer.seekTo(time)
    }

    fun showControl(){
        panel.callOnClick()
    }

    private fun onPlayButtonPressed() {
        if (isPlaying)
            youTubePlayer.pause()
        else
            youTubePlayer.play()
    }

    private fun updateState(state: PlayerConstants.PlayerState) {
        when (state) {
            PlayerConstants.PlayerState.ENDED -> isPlaying = false
            PlayerConstants.PlayerState.PAUSED -> isPlaying = false
            PlayerConstants.PlayerState.PLAYING -> isPlaying = true
            PlayerConstants.PlayerState.BUFFERING -> isPlaying = false
            else -> { }
        }

        updatePlayPauseButtonIcon(!isPlaying)
    }

    private fun updatePlayPauseButtonIcon(playing: Boolean) =
            playPauseButton.setImageResource(if (playing) R.drawable.ic_pause_24dp else R.drawable.ic_play_arrow_24dp)


    // YouTubePlayer callbacks

    override fun onStateChange(youTubePlayer: YouTubePlayer, state: PlayerConstants.PlayerState) {
        updateState(state)

        if (state === PlayerConstants.PlayerState.PLAYING || state === PlayerConstants.PlayerState.PAUSED || state === PlayerConstants.PlayerState.VIDEO_CUED) {
            panel.setBackgroundColor(ContextCompat.getColor(panel.context, android.R.color.transparent))
            progressBar.visibility = View.GONE

            if (isPlayPauseButtonEnabled) playPauseButton.visibility = View.VISIBLE

            updatePlayPauseButtonIcon(state === PlayerConstants.PlayerState.PLAYING)

        } else {
            updatePlayPauseButtonIcon(false)

            if (state === PlayerConstants.PlayerState.BUFFERING) {
                progressBar.visibility = View.VISIBLE
                panel.setBackgroundColor(ContextCompat.getColor(panel.context, android.R.color.transparent))
                if (isPlayPauseButtonEnabled) playPauseButton.visibility = View.INVISIBLE

            }

            if (state === PlayerConstants.PlayerState.UNSTARTED) {
                progressBar.visibility = View.GONE
                if (isPlayPauseButtonEnabled) playPauseButton.visibility = View.VISIBLE
            }
        }
    }
}
```

**Comment by SlayerMMD** [comment]
hey @sonusourav , is it happening for some set of urls or randomly occurring on any video?
Have not investigated it much. Jut started looking into it deeply now

**Comment by sonusourav** [comment]
> 
> 
> hey @sonusourav , is it happening for some set of urls or randomly occurring on any video?
> Have not investigated it much. Jut started looking into it deeply now

@SlayerMMD I don't think it has anything to do with videos

**Comment by sonusourav** [comment]
Hey @SlayerMMD wait, I think you are correct. Till now, I was checking with just one video id. Now, I checked with some other, and it seems to be working fine. Can others also check once

**Comment by SlayerMMD** [comment]
Hey @sonusourav , Yaa i knew for a fact that it works sometimes and sometimes it is failing, that's why wanted to be sure, it it failing for a particular video always? it will help debug the issue. But if it is randomly failing. It will be really hard to find the root cause

**Comment by sonusourav** [comment]
Hey @SlayerMMD , I can now confirmly say that there is something to do with video ids only because I tried to put the id which was not working in mine project to the sample project and its not working there also. (`0ZqiHrBW8Ug`) 

@PierfrancescoSoffritti Any idea why this might be happening?

**Comment by SlayerMMD** [comment]
Okay, So i debugged it further
We are getting this error when we are trying to use seekbar or directly using the seekTo method
Uncaught Error: Java exception was raised during method invocation line numner 106 source id https://www.youtube.com/
Now, i am not sure how we can check this out :D

**Comment by amitimg29** [comment]
hey guys, it works fine, when I load videos on landscape mode by default.

**Comment by PierfrancescoSoffritti** [comment]
@SlayerMMD is that error coming from the webview? if so you should have a look in the javascript code. Even though that "java exception" would indicate it's not coming from javascript.

**Comment by SlayerMMD** [comment]
Hey @PierfrancescoSoffritti , Yup i override the onConsoleMessage of the webchromeclient and when it gets stuck, we always see this error.

**Comment by singhaayush** [comment]
Hey Guys! seems this issue is auto-fixed for me from remote youtube scripts . Can you guys confirm the same . If it is resolved for all of you we could close this issue .

**Comment by sonusourav** [comment]
Yeah, even I was surprised to see it fixed all of a sudden 😅 . But the good thing is its fixed 🎊

**Comment by PierfrancescoSoffritti** [comment]
Nice, glad to hear. Maybe they changed something in the IFrame player.
